### PR TITLE
chore(routes): Setup react router  (React-router-dom)[#187364797]

### PR DESCRIPTION
**Pull request by andremugabo**
This PR sets up React Router (react-router-dom) for the client application. It includes the installation and configuration of React Router, the creation of dummy routes for the login page and the home page (/), and the establishment of a components and views file structure.
1. Installation of React Router:
         - Install react-router-dom and its TypeScript definitions.
         - Command to use:
                 npm install react-router-dom @types/react-router-dom
2. Configuration of React Router:
         - Created a routes directory with an AppRoutes.tsx file to manage application routes
         - Configured AppRoutes with BrowserRouter, Routes, and Route components to handle navigation.
         - Added dummy routes for the home page (/) and the login page (/login).

### CODE REVIEW

**Review by shebz2023 — APPROVED** [review]
Great!

**Review by tuyishimehope — APPROVED** [review]
Rebase

File: src/routes/AppRoutes.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const AppRoutes: React.FC = () => {
+    return (
+        <Router>
+            <Header />
+            <Routes>
+                <Route path="/" element={<HomePage />} />
+                <Route path="/login" element={<LoginPage />} />
+                <Route path="*" element={<NotFoundPage />} />
+            </Routes>
+        </Router>
```
Comment (SergeInGithub) [inline src/routes/AppRoutes.tsx]: This is good but there's a new way for routing which separates the main app from routes. It also has features like data loading, deferred loading out of the box, etc... look into it, hint: it uses createBrowserRouter and createRoutesFromElements

File: src/routes/AppRoutes.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
```
Comment (SergeInGithub) [inline src/routes/AppRoutes.tsx]: rename this file to routes.tsx and export it * in the index.ts file that you have to create in the routes folder

File: src/views/HomePage.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
```
Comment (SergeInGithub) [inline src/views/HomePage.tsx]: Rename views to pages and export homepage as regular not default, then export it from the index file in the pages folder

File: src/App.tsx [hunk 4]
```diff
@@ -1,11 +1,8 @@
...
 
-function App() {
-  return (
-    <main className="bg-slate-900 h-screen flex items-center justify-center">
-      <h1 className="text-7xl text-white">Welcome to E-commerce</h1>
-    </main>
-  );
-}
+const App: React.FC = () => {
+    return <AppRoutes />;
```
Comment (SergeInGithub) [inline src/App.tsx]: use RouterProvider from react-router-dom instead

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Look into the changes suggested below @andremugabo

File: src/routes/routes.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { createBrowserRouter, createRoutesFromElements, Route } from 'react-router-dom';
+import App from '../App';
+import { HomePage, LoginPage, NotFoundPage } from '../pages';
+
+const router = createBrowserRouter(
+  createRoutesFromElements(
+    <Route path="/" element={<App />}>
```
Comment (SergeInGithub) [inline src/routes/routes.tsx]: Create a file called `RootLayout` in the layouts folder then export it in the index file there
```suggestion
    <Route path="/" element={<RootLayout />}>
```

File: src/routes/routes.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { createBrowserRouter, createRoutesFromElements, Route } from 'react-router-dom';
+import App from '../App';
+import { HomePage, LoginPage, NotFoundPage } from '../pages';
+
+const router = createBrowserRouter(
```
Comment (SergeInGithub) [inline src/routes/routes.tsx]:
```suggestion
export const router = createBrowserRouter(
```

File: src/routes/routes.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  createRoutesFromElements(
+    <Route path="/" element={<App />}>
+      <Route index element={<HomePage />} />
+      <Route path="login" element={<LoginPage />} />
+      <Route path="*" element={<NotFoundPage />} />
+    </Route>
+  )
+);
+
+export default router;
```
Comment (SergeInGithub) [inline src/routes/routes.tsx]:
```suggestion
```

File: src/routes/index.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+import router from './routes'; 
+
+export default router;
```
Comment (SergeInGithub) [inline src/routes/index.ts]:
```suggestion
export * from './routes';
```

File: src/pages/index.ts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+export { HomePage } from './Home';
+export { default as LoginPage } from './LoginPage';
+export { default as NotFoundPage } from './NotFoundPage';
```
Comment (SergeInGithub) [inline src/pages/index.ts]:
```suggestion
export * from './Home';
export * from './LoginPage';
export * from './NotFoundPage';
```

File: src/pages/NotFoundPage.tsx [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+const NotFoundPage: React.FC = () => {
```
Comment (SergeInGithub) [inline src/pages/NotFoundPage.tsx]:
```suggestion
export function NotFoundPage() {
```

File: src/pages/NotFoundPage.tsx [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+        <div className="flex flex-col items-center justify-center h-screen text-center">
+            <h1 className="text-4xl mb-4">404 - Page Not Found</h1>
+            <Link to="/" className="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-700">
+                Back to Home
+            </Link>
+        </div>
+    );
+};
+
+export default NotFoundPage;
```
Comment (SergeInGithub) [inline src/pages/NotFoundPage.tsx]:
```suggestion
```

File: src/pages/LoginPage.tsx [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const LoginPage: React.FC = () => {
```
Comment (SergeInGithub) [inline src/pages/LoginPage.tsx]:
```suggestion
export function LoginPage() {
```

File: src/pages/LoginPage.tsx [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+    return (
+        <div className="flex items-center justify-center h-screen">
+          <h1 className="text-4xl">Login Page</h1>
+        </div>
+      );
+};
+
+export default LoginPage;
```
Comment (SergeInGithub) [inline src/pages/LoginPage.tsx]:
```suggestion
```

File: src/pages/Home.tsx [hunk 14]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export const HomePage: React.FC = () => {
```
Comment (SergeInGithub) [inline src/pages/Home.tsx]:
```suggestion
export function HomePage() {
```

File: src/main.tsx [hunk 15]
```diff
@@ -1,10 +1,15 @@
...
+import './index.css';
+import router from './routes'; 
 
-ReactDOM.createRoot(document.getElementById('root')!).render(
+const root = ReactDOM.createRoot(
+  document.getElementById('root') as HTMLElement
+);
+
+root.render(
   <React.StrictMode>
```
Comment (SergeInGithub) [inline src/main.tsx]:
```suggestion
ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
);
```

File: src/components/Header.tsx [hunk 16]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+const Header: React.FC = () => {
```
Comment (SergeInGithub) [inline src/components/Header.tsx]:
```suggestion
export const Header: React.FC = () => {
```

File: src/components/Header.tsx [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+                </div>
+                <Link to="/login" className="bg-white text-blue-500 px-4 py-2 rounded hover:bg-gray-200">
+                    Login
+                </Link>
+            </nav>
+        </header>
+    );
+};
+
+export default Header;
```
Comment (SergeInGithub) [inline src/components/Header.tsx]:
```suggestion
```

File: src/components/Header.tsx [hunk 18]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
```
Comment (SergeInGithub) [inline src/components/Header.tsx:3]: create an index file in the components folder and export like i've shown you on other index files below

File: src/App.tsx [hunk 19]
```diff
@@ -1,16 +1,14 @@
...
   return (
-    <Provider store={store}>
-      <main className="bg-slate-900 h-screen flex items-center justify-center">
-        <h1 className="text-7xl text-white">Welcome to E-commerce</h1>
-      </main>
-    </Provider>
+    <div>
+      <Header />
+      <Outlet />
+    </div>
```
Comment (SergeInGithub) [inline src/App.tsx]:
```suggestion
    <Provider store={store}>
        <RouterProvider router={router} />
    </Provider>
```

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Address the changes below @andremugabo

**Review by SergeInGithub — APPROVED** [review]
Nice work @andremugabo

**Pull request merged**